### PR TITLE
feat(developer): provide line number for some kmw compiler messages

### LINKED ISSUE #9931: feat(developer): Improve error message for keywords not supported on some platforms
Currently web doesn't support the `return` statement (#1092). If you try to compile a .kmn file that contains `return`, you get output similar to:

```
test_deadkeys.kmn - info KM05002: Building test_deadkeys.kmn
test_deadkeys.kmn - error KM02055: Statement return is not currently supported in output
test_deadkeys.kmn - info KM05007: test_deadkeys.kmn failed to build.
```

It would be nice if it would tell the line number where the error happens and the platform that isn't supported. Also the keyword could be highlighted:

```
error KM02055: Statement `return` is not currently supported in output for target `web`
```

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #9931.

Provide additional line and file context for some KeymanWeb compiler messages. This change requires debug data in the intermediate kmx data provided to the kmw compiler, which should have no impact on the final .js if `options.debug==false`.

The following messages have had line number context added:
* `Error_NotSupportedInKeymanWebContext`
* `Error_NotSupportedInKeymanWebOutput`
* `Error_VirtualCharacterKeysNotSupportedInKeymanWeb`
* `Error_VirtualKeysNotValidForMnemonicLayouts`
* `Warn_ExtendedShiftFlagsNotSupportedInKeymanWeb`
* `Hint_UnreachableKeyCode`
* `Warn_HelpFileMissing`
* `Warn_EmbedJsFileMissing`
* `Error_NotAnyRequiresVersion14`

Line number metadata is not easily accessible where the following messages are generated, so these are a potential future improvement:
* `Warn_OptionStoreNameInvalid`
* `Error_NotSupportedInKeymanWebStore`

A unit test has been added for `Error_NotSupportedInKeymanWebOutput`, but not for the other messages.

Potential future improvements to unit tests:
* Tests for the other messages listed here
* Tests for line number correctness

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc-kmn/test/test-messages.ts [hunk 1]
```diff
@@ -80,4 +80,11 @@ describe('CompilerMessages', function () {
     assert.equal(callbacks.messages[0].message, "A key on layer \"default\" has no identifier.");
   });
 
+  // ERROR_NotSupportedInKeymanWebOutput
+
+  it('should generate ERROR_NotSupportedInKeymanWebOutput if a rule has `return` in the output', async function() {
+    await testForMessage(this, ['invalid-keyboards', 'error_not_supported_in_keyman_web_output.kmn'], KmnCompilerMessages.ERROR_NotSupportedInKeymanWebOutput);
+    assert.equal(callbacks.messages[0].message, "Statement 'return' is not currently supported in output for web and touch targets");
+  });
```
Comment (darcywong00) [inline developer/src/kmc-kmn/test/test-messages.ts:88]: Can we assert line 11 in the kmn is also reported in the compiler message?

Comment (mcdurdin) [reply developer/src/kmc-kmn/test/test-messages.ts:88]: See topmost comment:

>Potential future improvements to unit tests:
>
>Tests for the other messages listed here
>Tests for line number correctness

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.206-alpha](https://keyman.com/downloads/releases/alpha/17.0.206)

**Pull request merged**